File: UL/System/Boolean.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace System
{
    public struct Boolean
    {
        public static readonly string FalseString = "false";
        public static readonly string TrueString = "true";

        public static bool Parse(string value)
        {
            if(value == null)
            {
                throw new ArgumentNullException();
            }
            if (value == TrueString)
                return true;
            else if(value == FalseString)
                return false;
            else
            {
                throw new FormatException(value);
            }
        } 

        public extern override string ToString();

        public static bool TryParse(string value,out bool v)
        {
            try
            {
                v =  Parse(value);
                return true;
            }
            catch(Exception e)
            {
                v = false;
                return false;
            }
        }
    }
}

File: ul/CppApp/Boolean.cpp
#include "stdafx.h"
#include "Boolean.h"
#include "Object.h"
#include "String.h"
#include "ArgumentNullException.h"
#include "FormatException.h"
#include "Exception.h"
Ref<System::String> System::Boolean::FalseString;
Ref<System::String> System::Boolean::TrueString;
System::Boolean System::Boolean::Parse(Ref<System::String>  value)
{
	if(value == nullptr)
	{
		throw new System::ArgumentNullException();
	}
	if(value == TrueString)
		return true;
	else
		if(value == FalseString)
			return false;
		else
		{
			throw new System::FormatException(value);
		}
}
System::Boolean System::Boolean::TryParse(Ref<System::String>  value,System::Boolean & v)
{
	try
	{
		v = Parse(value);
		return true;
	}
	catch(System::Exception e)
	{
		v = false;
		return false;
	}
}
